### PR TITLE
Replace refcat2's XMatch Gaia-ID lookup with local coordinate-based IDs

### PR DESCRIPTION
**Pull request by mwcraig**
## Summary

`refcat2()` attached a Gaia DR2 source_id to each star via CDS XMatch. Unlike the Vizier cone searches — which fall back across several mirror servers — the XMatch service runs only in Strasbourg with no mirrors, and a recent outage (502s) broke `refcat2()` outright. The crossmatch also silently dropped any star that failed to Gaia-match.

Nothing downstream depends on the id being a Gaia ID: the `id` column is only used as a pivot index in `CatalogData.passband_columns` and is deleted in the photometry pipeline. #610 deferred the "is the Gaia ID worth the extra query?" decision; this PR answers no.

## Changes

- `refcat2()` now generates IAU-style coordinate designations (`REFCAT2SP J+RRR.RRRR±DD.DDDD`) locally, the same approach `apass_dr9` already uses (`APASSSP J...`). The designation code is extracted into a shared `_iau_designation_ids()` helper called by both fetchers.
- `_attach_gaia_ids()`, the XMatch query, and the associated imports are deleted.
- The two quality filters (galaxy rejection via masked `e_pmRA`/`e_pmDE`, non-Gaia-object rejection via masked `e_Gmag`) are unchanged, so the catalog contains exactly the same stars as before; only the `id` column changes.

User-visible effects:
- The id format changes from a Gaia DR2 source_id to a coordinate designation.
- Stars are no longer silently dropped when the Gaia crossmatch misses — nothing is dropped anymore.

## Tests

- New unit tests for `_iau_designation_ids` cover the exact format, sign/padding edge cases, and reproduce the existing APASS designation.
- The monkeypatched XMatch upload test is replaced by a fully offline `test_process_refcat2_filters_and_ids` exercising both filters and the generated ids.
- The remote-data `test_find_refcat2` spot check now expects EY UMa's coordinate designation instead of its Gaia source_id.
- `pytest stellarphot/tests/test_catalogs.py -k refcat2 --remote-data` passes against live Vizier **while XMatch is down**, which is the point of the change.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01YYsY9Meu2TY2LbLLVzZ3Pj

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -28,6 +28,13 @@ Other Changes and Additions
   They remain available as top-level names (``from stellarphot import
   apass_dr9``) and, for one or two releases, from ``stellarphot.core`` via a
   back-compat shim that raises an ``AstropyDeprecationWarning``. [#194]
++ ``refcat2`` IDs are now locally generated IAU-style coordinate designations
+  (``REFCAT2SP J+RRR.RRRR±DD.DDDD``) instead of Gaia DR2 source_ids fetched
+  from CDS XMatch. The id format changes, and stars are no longer silently
+  dropped when the Gaia crossmatch misses (nothing is dropped anymore). The
+  XMatch service has no mirror — unlike the Vizier queries, which fall back
+  across several servers — and an XMatch outage previously broke ``refcat2()``
+  entirely.
```
[the remark this answers is not in the document]
Comment (mwcraig) [reply CHANGES.rst]: Good catch. The #586 bullet later in this same unreleased 2.0.0 section describes the XMatch row-index join that this PR removes, so as written the release notes would contradict themselves. Since neither entry has shipped, the right fix is to drop the #586 bullet (the bug it fixed no longer exists — there is no XMatch query left to join back) and mention #586 as superseded in this entry. Will update the PR accordingly.

File: stellarphot/tests/test_catalogs.py [hunk 2]
```diff
@@ -266,101 +269,51 @@ def test_find_refcat2(mag_limit, mag_limit_band):
     for band in ["GBP", "GRP", "GG", "SG", "SR", "SI", "SZ", "J", "H", "K"]:
         assert band in all_refcat2["passband"]
 
-    # The Gaia IDs must be unique per star. The catalog is tidy (one row per
+    # The IDs must be unique per star. The catalog is tidy (one row per
     # star per passband), so compare the number of distinct IDs to the number
     # of distinct positions.
     n_stars = len(set(ra.value for ra in all_refcat2["ra"]))
     assert len(set(all_refcat2["id"])) == n_stars
```
[the remark this answers is not in the document]
Comment (mwcraig) [reply stellarphot/tests/test_catalogs.py]: Fair point — the comment says "distinct positions" but the code counts distinct RAs only. Two things worth noting: this line is pre-existing (the PR only reworded the surrounding comment from "Gaia IDs" to "IDs"), and for this fixed test field RA alone happens to be unique per star (the expected-data file has 187 distinct RAs across 187 stars, verified locally), so the assertion behaves as intended today. That said, counting distinct `(ra, dec)` pairs is more robust and makes the code match the comment, so I'll tighten it in this PR.

File: CHANGES.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (mwcraig) [reply CHANGES.rst]: Done in 5da1cc7 — the #586 bullet is removed and the new entry notes it supersedes that fix.

File: stellarphot/tests/test_catalogs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mwcraig) [reply stellarphot/tests/test_catalogs.py]: Done in 5da1cc7 — `n_stars` now counts distinct `(ra, dec)` pairs. Remote-data refcat2 tests still pass.
